addons/portal/controllers/portal_thread.py

```diff
@@ -69,11 +69,15 @@ def portal_message_fetch(self, thread_model, thread_id, fetch_params=None, **kw)
         # Check access
         Message = request.env['mail.message']
         if kw.get('token'):
-            access_as_sudo = ThreadController._get_thread_with_access(
+            thread = ThreadController._get_thread_with_access(
```

Comment:
Why do we only check _get_thread_with_access if a token is provided? Shouldn't we check in all cases?

Do we currently rely on _message_fetch failing/returning nothing if the thread is not accessible?

Reply:
Yes, we rely on ACL failure in non token cases. It's the old way in Portal. With the replacement of the fetch route, we check for access to the thread and all cases first as the other routes.